GpsDataCapturer/sharedlibrary/src/main/java/com/google/sharedlibrary/FusedLocationProviderHelper.java

```diff
@@ -0,0 +1,85 @@
+package com.google.sharedlibrary;
+
+import android.annotation.SuppressLint;
+import android.os.Looper;
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+
+import com.google.android.gms.location.FusedLocationProviderClient;
+import com.google.android.gms.location.LocationCallback;
+import com.google.android.gms.location.LocationRequest;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.Task;
+
+/**
+ * Helper class that help the GpsDataCaptureService to start/stop the FusedLocationProviderClient.
+ */
+public class FusedLocationProviderHelper {
+    private static final String TAG = "FLPHelper";
+    private static final int INTERVAL = 1000;
+    private static final int FASTEST_INTERVAL = 1000;
+    private static LocationRequest locationRequest;
+
+    /**
+     * Start fusedLocationProviderClient
+     */
+    @SuppressLint("MissingPermission")
+    public static void startFusedLocationProviderClient(
+            FusedLocationProviderClient fusedLocationProviderClient,
+            LocationCallback locationCallback) {
+        createLocationRequest();
+
+        fusedLocationProviderClient.requestLocationUpdates(locationRequest, locationCallback,
```

Comment:
locationRequest should be a local variable here :
locationRequest = createLocationRequest(); 

or drop local and pass createLocationRequest() directly to requestLocationUpdates